spec/models/widget_spec.rb

```diff
@@ -955,15 +955,19 @@
   end
 
   describe ".paper_trail.update_columns", versioning: true do
-    let(:widget) { described_class.create! name: "Bob", an_integer: 1 }
-
     it "creates a version record" do
-      widget = described_class.create
+      widget = described_class.create(updated_at: "2015-01-01 15:00")
       expect(widget.versions.count).to eq(1)
-      widget.paper_trail.update_columns(name: "Bugle")
+      now = nil
+      freeze_time do
+        now = Time.zone.now
+        widget.paper_trail.update_columns(name: "Bugle")
+      end
+
       expect(widget.versions.count).to eq(2)
       expect(widget.versions.last.event).to(eq("update"))
       expect(widget.versions.last.changeset[:name]).to eq([nil, "Bugle"])
+      expect(widget.versions.last.created_at.to_i).to eq(now.to_i)
```

Comment:
I'm not familiar with `freeze_time`. Can we do something like `expect {}.to_not(change { widget.versions.last.created_at })` instead? Would that achieve the same testing goal?

Reply:
We want to test that `widget.versions.last.created_at`  will change to `Time.now` (current time) instead of `widget.updated_at` (record last update date) (it will always change)

Alternatively (without freezing time and rounding instead - not perfect) I think we could do:

- option 1 (consistent with existing spec):
```diff
     it "creates a version record" do
       widget = described_class.create(updated_at: "2015-01-01 15:00")
       expect(widget.versions.count).to eq(1)
-      now = nil
-      freeze_time do
-        now = Time.zone.now
-        widget.paper_trail.update_columns(name: "Bugle")
-      end
+      widget.paper_trail.update_columns(name: "Bugle")
 
       expect(widget.versions.count).to eq(2)
       expect(widget.versions.last.event).to(eq("update"))
       expect(widget.versions.last.changeset[:name]).to eq([nil, "Bugle"])
-      expect(widget.versions.last.created_at.to_i).to eq(now.to_i)
+      expect(widget.versions.last.created_at).to be_within(1.second).of(Time.zone.now)
     end
end
```

using the `change` matcher with `widget.versions.last` is a bit complicated because the subject (`versions.last`) of the test will be different for comparison, but I guess we could do something like this:

- option 2 (using `change` matcher and a separate scenario):
```diff
   describe ".paper_trail.update_columns", versioning: true do
     it "creates a version record" do
-      widget = described_class.create(updated_at: "2015-01-01 15:00")
+      widget = described_class.create
       expect(widget.versions.count).to eq(1)
-      now = nil
-      freeze_time do
-        now = Time.zone.now
-        widget.paper_trail.update_columns(name: "Bugle")
-      end
+      widget.paper_trail.update_columns(name: "Bugle")
 
       expect(widget.versions.count).to eq(2)
       expect(widget.versions.last.event).to(eq("update"))
       expect(widget.versions.last.changeset[:name]).to eq([nil, "Bugle"])
-      expect(widget.versions.last.created_at.to_i).to eq(now.to_i)
+    end
+
+    it "uses current time for the version created_at" do
+      widget = described_class.create(updated_at: "2015-01-01 15:00")
+
+      expect { widget.paper_trail.update_columns(name: "Bugle") }.to(
+        change {
+          Time.now.to_f - widget.versions.where(event: "update").last&.created_at.to_f < 1
+        }.to(eq(true))
+      )
     end
   end
```

Reply:
Thanks for the options! I think I'll go with option 2.